lego/apps/polls/models.py

```diff
@@ -0,0 +1,61 @@
+from django.conf import settings
+from django.db import models
+from django.db.models import Sum
+from django.utils import timezone
+
+from lego.apps.content.models import Content
+from lego.apps.polls.permissions import PollPermissionHandler
+from lego.apps.users.models import User
+from lego.utils.models import BasisModel
+
+
+def get_time_delta():
+    return timezone.now() + timezone.timedelta(weeks=52)
+
+
+class Poll(Content, BasisModel):
+
+    description = models.TextField(blank=True)
+
+    valid_until = models.DateTimeField(default=get_time_delta)
+
+    answered_users = models.ManyToManyField(User, related_name="answered_polls")
+
+    def __str__(self):
+        return self.title
+
+    @property
+    def total_votes(self):
+        return Option.objects.filter(poll=self).aggregate(total_votes=Sum("votes"))[
+            "total_votes"
+        ]
+
+    def get_absolute_url(self):
+        return f"{settings.FRONTEND_URL}/polls/{self.id}/"
+
+    def vote(self, user, option_id):
+        if user in self.answered_users.all():
+            return "forbidden"
```

Comment:
Instead of hardcoding this, it is better to use proper exceptions:

https://github.com/webkom/lego/blob/474ae2536507e304a87e007e542e0710ba24b57e/lego/apps/users/views/student_confirmation.py#L33